pkg/config/validation.go

```diff
@@ -209,30 +209,44 @@ func validatePodIntegrationOptions(c *configapi.Configuration) field.ErrorList {
 		return allErrs
 	}
 
-	if c.Integrations.PodOptions == nil {
-		return field.ErrorList{field.Required(podOptionsPath, "cannot be empty when pod integration is enabled")}
-	}
-	if c.Integrations.PodOptions.NamespaceSelector == nil {
-		return field.ErrorList{field.Required(namespaceSelectorPath, "a namespace selector is required")}
+	validateNamespaceSelector := func(namespaceSelector *metav1.LabelSelector, namespaceSelectorPath *field.Path, allErrs field.ErrorList) field.ErrorList {
+		allErrs = append(allErrs, validation.ValidateLabelSelector(namespaceSelector, validation.LabelSelectorValidationOptions{}, namespaceSelectorPath)...)
+		selector, err := metav1.LabelSelectorAsSelector(namespaceSelector)
+		if err != nil {
+			return allErrs
+		}
+		prohibitedNamespaces := []labels.Set{{corev1.LabelMetadataName: metav1.NamespaceSystem}}
+		if c.Namespace != nil && *c.Namespace != "" {
+			prohibitedNamespaces = append(prohibitedNamespaces, labels.Set{corev1.LabelMetadataName: *c.Namespace})
+		}
+		for _, pn := range prohibitedNamespaces {
+			if selector.Matches(pn) {
+				allErrs = append(allErrs, field.Invalid(namespaceSelectorPath, namespaceSelector,
+					fmt.Sprintf("should not match the %q namespace", pn[corev1.LabelMetadataName])))
+			}
+		}
+		return allErrs
 	}
 
-	prohibitedNamespaces := []labels.Set{{corev1.LabelMetadataName: metav1.NamespaceSystem}}
-
-	if c.Namespace != nil && *c.Namespace != "" {
-		prohibitedNamespaces = append(prohibitedNamespaces, labels.Set{corev1.LabelMetadataName: *c.Namespace})
+	// At least one namespace selector must be non-nil and enabled.
+	// It is ok for both to be non-nil; pods will only be managed if all non-nil selectors match
+	hasNamespaceSelector := false
+	if features.Enabled(features.ManagedJobsNamespaceSelector) {
+		if c.ManagedJobsNamespaceSelector != nil {
```

Comment:
```suggestion
	if features.Enabled(features.ManagedJobsNamespaceSelector) && c.ManagedJobsNamespaceSelector != nil{
```

This could be minimized.

Reply:
done